govcd/disk.go

```diff
@@ -227,13 +227,20 @@ func (disk *Disk) Delete() (Task, error) {
 func (disk *Disk) Refresh() error {
 	util.Logger.Printf("[TRACE] Disk refresh, HREF: %s\n", disk.Disk.HREF)
 
-	fetchedDisk, err := FindDiskByHREF(disk.client, disk.Disk.HREF)
+	if *disk == (Disk{}) {
```

Comment:
This code can fail easily.  I prefer the following.
```suggestion
	if disk.Disk != nil && disk.Disk.HREF != "" {
```

Reply:
changed